packages/mobile/src/home/WalletHome.tsx

```diff
@@ -160,6 +180,14 @@ function WalletHome() {
       />
       <SendOrRequestBar />
       {shouldShowCashInBottomSheet() && <CashInBottomSheet />}
+      {showRecentDapps && (
```

Comment:
any ideas on how else i could do this? ideally i want to keep the dapps logic together (i.e. have this bottom sheet inside the `RecentlyUsedDapps` component) but I could not get the bottom sheet to be on the bottom of the screen with the way flex styles work since `RecentlyUsedDapps` is inside the section list 🥲

Reply:
Yup, I think it's fine. One possible nit is:

Right now, the `useOpenDapp` hook and `DAppsBottomSheet` components are implicitly connected since the component uses the state from the hook so this component (and `DAppsExplorerScreen`) is basically acting as a middle man.

Something we _could_ do (not really necessary, just thinking out load) is to make `useOpenDapp` return the component and just render it here. In that case, `useOpenDapp` would just return the existing `onSelectDapp` function and a new `bottomSheetDialog` component. And then here we would just render that `bottomSheetDialog`. That way there are less props being returned and passed to the bottom sheet here.

Again, this is very nit-picky, I think the current solution is perfectly fine 😄

Reply:
i was actually thinking about this too, but didn't do it because it didn't feel "react-y" to return a component from a hook. it felt very similar to get props from a hook then pass them to a component, vs get component from hook and put it in the render function.

but now that you mentioned it, i tried it and it does save exposing a bunch of functions/state for opening the dapps bottom sheet. and i don't really know why i feel like it is "un-react" to return a component from a hook, i couldn't find a good reason online so it's probably just fine! let me know if you like how this looks?

Reply:
@jeanregisser maybe you have some thoughts about this too?

i think returning components from a hook is not a well liked pattern because it is hard to keep the returned component's identity the same and there are issues with react mounting/unmounting the component unnnecessarily. i think that is not a big problem here, and i've tried to reduce the impact with useMemo but still not feeling great about it (or my previous solution of having the hook provide the props for the bottom sheet...)

Reply:
It looks ok for now.
But I think we should consider turning BottomSheets (and Dialogs) as screens handled by the navigator.
This gives greater flexibility to really render fullscreen and trigger them from anywhere.
Avoiding the issue seen here and others we have (see https://github.com/valora-inc/wallet/issues/881).